src/core/settings/qgssettingsentry.cpp

```diff
@@ -55,37 +55,89 @@ QString QgsSettingsEntryBase::key( const QStringList &dynamicKeyPartList ) const
   QString completeKey = mKey;
   if ( !mPluginName.isEmpty() )
   {
-    if ( !completeKey.startsWith( "/" ) )
-      completeKey.prepend( "/" );
+    if ( !completeKey.startsWith( '/' ) )
+      completeKey.prepend( '/' );
     completeKey.prepend( mPluginName );
   }
 
   if ( dynamicKeyPartList.isEmpty() )
   {
     if ( hasDynamicKey() )
-      QgsLogger::warning( QStringLiteral( "Settings '%1' have a dynamic key but the dynamic key part was not provided" ).arg( completeKey ) );
+      QgsDebugMsg( QStringLiteral( "Settings '%1' have a dynamic key but the dynamic key part was not provided" ).arg( completeKey ) );
 
     return completeKey;
   }
   else
   {
     if ( !hasDynamicKey() )
     {
-      QgsLogger::warning( QStringLiteral( "Settings '%1' don't have a dynamic key, the provided dynamic key part will be ignored" ).arg( completeKey ) );
+      QgsDebugMsg( QStringLiteral( "Settings '%1' don't have a dynamic key, the provided dynamic key part will be ignored" ).arg( completeKey ) );
       return completeKey;
     }
 
     for ( int i = 0; i < dynamicKeyPartList.size(); i++ )
     {
-      completeKey.replace( QString( "%%1" ).arg( QString::number( i + 1 ) ), dynamicKeyPartList.at( i ) );
+      completeKey.replace( QStringLiteral( "%%1" ).arg( QString::number( i + 1 ) ), dynamicKeyPartList.at( i ) );
```

Comment:
is this double `%%` intended?

Reply:
The `%%` is intended, to get `%1`, `%2`, `%3`,... in the end to replace in the key string. Maybe it would be more clear if I replace it with `QStringLiteral( "%" ).append( QString::number( i + 1 ) )`